pkg/docker/config/config.go

```diff
@@ -46,13 +46,14 @@ func SetAuthentication(sys *types.SystemContext, registry, username, password st
 			return false, setAuthToCredHelper(ch, registry, username, password)
 		}
 
-		err := setAuthToKernelKeyring(registry, username, password)
-		if err == nil {
-			logrus.Debugf("credentials for (%s, %s) were stored in the kernel keyring\n", registry, username)
-			return false, nil
+		if sys.AuthFilePath == "" {
```

Comment:
Please add a comment here. It's a bit a of an inverted logic to achieve the fall-through and readers of the code might get stuck here.

What could help is describing the intended behaviour:
* The keyring might not work in all environments (e.g., missing capability) and isn't supported on all platforms. Hence, we want to fall-back to using the authfile in case the keyring failed.
* However, if the `sys.AuthFilePath` is set, we want adhere to the user specification and not use the keyring.